mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,595 @@
+                               ElectrumBlockHeader, ElectrumClient, ElectrumNonce, EstimateFeeMethod,
+                               UtxoRpcClientEnum, UtxoRpcError};
+use crate::utxo::utxo_standard::UtxoStandardCoin;
+use crate::{MarketCoinOps, MmCoin};
+use bitcoin::blockdata::block::BlockHeader;
+use bitcoin::blockdata::script::Script;
+use bitcoin::blockdata::transaction::Transaction;
+use bitcoin::consensus::encode::{deserialize, serialize_hex};
+use bitcoin::hash_types::{BlockHash, TxMerkleNode, Txid};
+use bitcoin_hashes::{sha256d, Hash};
+use common::executor::{spawn, Timer};
+use common::jsonrpc_client::JsonRpcErrorType;
+use common::log;
+use derive_more::Display;
+use futures::compat::Future01CompatExt;
+use keys::hash::H256;
+use lightning::chain::{chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator},
+                       Confirm, Filter, WatchedOutput};
+use rpc::v1::types::H256 as H256Json;
+use std::cmp;
+use std::convert::{TryFrom, TryInto};
+
+const CHECK_FOR_NEW_BEST_BLOCK_INTERVAL: u64 = 60;
+const MIN_ALLOWED_FEE_PER_1000_WEIGHT: u32 = 253;
+
+#[derive(Debug, Display)]
+pub enum FindWatchedOutputSpendError {
+    #[display(fmt = "Can't convert transaction: {}", _0)]
+    TransactionConvertionErr(String),
+    #[display(fmt = "Can't deserialize block header: {}", _0)]
+    BlockHeaderDeserializeErr(String),
+}
+
+async fn find_watched_output_spend_with_header(
+    electrum_client: &ElectrumClient,
+    output: &WatchedOutput,
+) -> Result<Option<(BlockHeader, usize, Transaction, u64)>, FindWatchedOutputSpendError> {
+    // from_block parameter is not used in find_output_spend for electrum clients
+    let utxo_client: UtxoRpcClientEnum = electrum_client.clone().into();
+    let output_spend = match utxo_client
+        .find_output_spend(
+            H256::from(output.outpoint.txid.as_hash().into_inner()),
+            output.script_pubkey.as_ref(),
+            output.outpoint.index.into(),
+            BlockHashOrHeight::Hash(Default::default()),
+        )
+        .compat()
+        .await
+    {
+        Ok(Some(output)) => output,
+        _ => return Ok(None),
+    };
+
+    if let BlockHashOrHeight::Height(height) = output_spend.spent_in_block {
+        if let Ok(header) = electrum_client.blockchain_block_header(height as u64).compat().await {
+            match deserialize(&header) {
+                Ok(h) => {
+                    let spending_tx = match Transaction::try_from(output_spend.spending_tx) {
+                        Ok(tx) => tx,
+                        Err(e) => return Err(FindWatchedOutputSpendError::TransactionConvertionErr(e.to_string())),
+                    };
+                    return Ok(Some((h, output_spend.input_index, spending_tx, height as u64)));
+                },
+                Err(e) => return Err(FindWatchedOutputSpendError::BlockHeaderDeserializeErr(e.to_string())),
+            }
+        }
+    }
```

Comment:
This deserves a separate function I think

Reply:
Simplified it instead of creating a new function since this block of code is used here only.